src/components/databaseConnect.ts

```diff
@@ -0,0 +1,73 @@
+import PostgresDB from './postgresConnect'
+import GraphQLdb from './graphQLConnect'
+
+class DBConnect {
+  private static _instance: DBConnect
+
+  constructor() {}
+
+  public static get Instance() {
+    return this._instance || (this._instance = new this())
+  }
+
+  public query = PostgresDB.query
```

Comment:
should we explicitly say this is pgQuery ?

Reply:
Not sure what you mean here. 🤷‍♂️

Reply:
If it's a single class encompassing both Postgres and GraphQL, and if it exposes query method, we probably want to differentiate if it's query method for postgres or GraphQL

Reply:
Okay, I see what you mean now. Only problem is that the DBConnect object that is passed to the evaluator requires a `.query` method on it to match the type requirements of the `node-postgres` connection object used by the evaluator.

Don't see a straightforward way around this right now.

Reply:
Why would evaluator need databaseConnect ? It should just get a singleton of Postgres and Graphql

To be honest I think we need to be quite clear about databaseConnect, and what it's there for. Would have preferred it to be in a seperate issue, so we can discuss what it does and why create it. 

If it's an abstraction, so we can combine both PG and GQL queries, then it should really be named something like dbQueryHelper. In which case we put actual logic of the queries in that class. And in that case Postgres and Graphql singletons are there to abstract connection and basic query logic.

Reply:
Sure, yeah sorry, I should have discussed it before shoving it into this issue. It's just that I was wanting to write a GraphQL query instead of a complicated SQL query to get the templateIds of some of the more distant tables, and asked Nicole if it'd be okay to make some of the methods in PostgresConnect use GraphQL and she wasn't keen to mix-and-match and suggested this approach. In the end I didn't actually write the GraphQL queries yet (haven't implemented triggers on those tables), but I'm happy to re-do this if we want.

The evaluator doesn't need DBConnect per se, it just needs a valid PostGres connection (with a `.query` method, and a GraphQL object (with endpoint, fetch, etc.) if it's using those operators. DBConnect fits the bill for both, and the calling method (e.g. TriggersAndActions) will have only the DBConnect object available to pass in.

Shall I make a new issue to address this, or shall we have a chat about it first?